streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1452,4 +1459,24 @@ private RuntimeException performOnStreamTasks(final StreamTaskAction action) {
         return firstException;
     }
 
+    /**
+     * Return information about the current {@link StreamThread}.
+     *
+     * @return {@link ThreadMetadata}.
+     */
+    public synchronized ThreadMetadata threadMetadata() {
+        return threadMetadata;
+    }
+
+    private synchronized void updateThreadMetadata() {
+        Set<TaskMetadata> activeTasksMetadata = new HashSet<>();
```

Comment:
`final` is for the var `activeTasksMetadata` (not for the method). We try apply a "use `final` whenever possible" policy. It's just some nit.